napari/_vispy/canvas.py

```diff
@@ -1,63 +1,193 @@
+        self._scene_canvas.context.set_depth_func('lequal')
+
+        # Connecting events from SceneCanvas
+        self._scene_canvas.events.key_press.connect(
+            self._key_map_handler.on_key_press
+        )
+        self._scene_canvas.events.key_release.connect(
+            self._key_map_handler.on_key_release
+        )
+        self._scene_canvas.events.draw.connect(self.enable_dims_play)
+        self._scene_canvas.events.draw.connect(self.vispy_camera.on_draw)
+
+        self._scene_canvas.events.mouse_double_click.connect(
+            self._on_mouse_double_click
+        )
+        self._scene_canvas.events.mouse_move.connect(self._on_mouse_move)
+        self._scene_canvas.events.mouse_press.connect(self._on_mouse_press)
+        self._scene_canvas.events.mouse_release.connect(self._on_mouse_release)
+        self._scene_canvas.events.mouse_wheel.connect(self._on_mouse_wheel)
+        self._scene_canvas.events.resize.connect(self.on_resize)
+        self._scene_canvas.events.draw.connect(self.on_draw)
+        self.viewer.cursor.events.style.connect(self._on_cursor)
+        self.viewer.cursor.events.size.connect(self._on_cursor)
+        self.viewer.events.theme.connect(self._on_theme_change)
+        self.viewer.camera.events.interactive.connect(self._on_interactive)
+        self.viewer.layers.events.reordered.connect(self._reorder_layers)
+        self.viewer.layers.events.removed.connect(self._remove_layer)
+        self.destroyed.connect(self._disconnect_theme)
+
+    @property
+    def destroyed(self) -> pyqtBoundSignal:
+        return self._scene_canvas._backend.destroyed
+
+    @property
+    def native(self):
+        """Returns the native widget of the Vispy SceneCanvas."""
+        return self._scene_canvas.native
 
     @property
-    def destroyed(self):
-        return self._backend.destroyed
+    def screen_changed(self):
+        """Returning signal indicating whether the window screen has changed."""
+        return self._scene_canvas._backend.screen_changed
 
     @property
-    def background_color_override(self):
-        return self._background_color_override
+    def background_color_override(self) -> str:
+        """Background color of VispyCanvas.view returned as hex string. When not None, color is shown instead of
+        VispyCanvas.bgcolor. The setter expects str (any in vispy.color.get_color_names) or hex starting
+        with # or a tuple | np.array ({3,4},) with values between 0 and 1.
+
+        """
+        if self.view in self.central_widget._widgets:
+            return self.view.bgcolor.hex
 
     @background_color_override.setter
-    def background_color_override(self, value):
-        self._background_color_override = value
-        self.bgcolor = value or self._last_theme_color
+    def background_color_override(
+        self, value: Union[str, npt.ArrayLike]
+    ) -> None:
+        if self.view in self.central_widget._widgets:
+            self.view.bgcolor = value
```

Comment:
Not sure I understand this code; can you explain it?

Reply:
Ah this has to do with one of the next steps regarding multichannel (multiple views instead of just 1), but will revert as that is not there yet.

Reply:
resolved with [0b874c9](https://github.com/napari/napari/pull/5432/commits/0b874c9005cef15c6308b50fdc25c0ebe5bd3a09)